collector/app.go

```diff
@@ -32,3 +32,25 @@ func GetAppsCollection(c *CollectorOpts, url string) *client.AppCollection {
 
 	return appCollection
 }
+
+func GetAllAppData(c *CollectorOpts, url string) []rancher.App {
+	appList := GetAppCollection(c, url)
+	if appList == nil {
+		return nil
+	}
+
+	appData := appList.Data
```

Comment:
Repeating this pattern so much feels kinda gross. Would be nice to [add it to norman](https://github.com/rancher/norman/blob/c032c4611f2eec1652ef37d254f0e8ccf90c80aa/generator/type_template.go#L101) if we are still updating that. I see some of them are customized, for those it makes sense to write, but otherwise it adds lots of boilerplace code that we may need to update or where bugs can surface. Wish we could do something with interfaces but don't see how thats possible.

Reply:
Agreed. I was thinking to use interfaces but as you don't see how thats possible.